app/src/main/java/com/main/mealplanner/MainActivity.kt

```diff
@@ -10,15 +10,13 @@ class MainActivity : AppCompatActivity() {
     private lateinit var mainFragment: MainFragment
 
     override fun onCreate(savedInstanceState: Bundle?) {
-        mainFragment = MainFragment.newInstance()
+        mainFragment = MainFragment()
```

Comment:
This is the only line of code that you need after removing the companion object that you had in the main fragment class. I think this really simplifies the process of instantiating the main fragment.